src/Element_TEST.cc

```diff
@@ -72,6 +72,82 @@ TEST(Element, Required)
   ASSERT_EQ(elem.GetRequired(), "1");
 }
 
+/////////////////////////////////////////////////
+TEST(Element, SetExplicitlySetInFile)
```

Comment:
What would the behavior be if `InsertElement` was called after `elem->SetExplicitlySetInFile(false)`? Could we have test cases for this as well?

Reply:
If that happens then the element inserted will have `ExplicitlySetInFile` set to true and it's parent would have to false (as it was not set in the file). Will add a test for that.

Reply:
Added at: https://github.com/osrf/sdformat/pull/542/commits/4fcfa21dbdfb07e5cefdd885430ca026e7fc796e